### PR TITLE
Refactor parts of Save.pm

### PR DESCRIPTION
**Pull request by Photon89**
I can't find the mistake here. I'm getting:

```
Trying to save file to /home/photon/Bilder/Selection_001.avif
*** unhandled exception in callback:
***   Can't use string ("avif") as a HASH ref while "strict refs" in use at /home/photon/github/shutter/share/shutter/resources/modules/Shutter/Pixbuf/Save.pm line 68.
***   	Shutter::Pixbuf::Save::set_quality_setting("avif") called at /home/photon/github/shutter/share/shutter/resources/modules/Shutter/Pixbuf/Save.pm line 152
***   	Shutter::Pixbuf::Save::save_pixbuf_to_file(Shutter::Pixbuf::Save=HASH(0x648aee8e9598), Gtk3::Gdk::Pixbuf=HASH(0x648aefa133a0), "/home/photon/Bilder/Selection_001.avif", "avif", 100) called at ./bin/shutter line 6453
***   	Shutter::App::fct_take_screenshot(Gtk3::ToolButton=HASH(0x648aef4cf148), "select", undef, undef) called at ./bin/shutter line 3046
***   	Shutter::App::__ANON__() called at /usr/lib/perl5/5.38/vendor_perl/Glib/Object/Introspection.pm line 67
***   	eval {...} called at /usr/lib/perl5/5.38/vendor_perl/Glib/Object/Introspection.pm line 67
***   	Glib::Object::Introspection::__ANON__(Shutter::App=HASH(0x648aee8d9618)) called at ./bin/shutter line 10986
***  ignoring at /usr/lib/perl5/5.38/vendor_perl/Glib/Object/Introspection.pm line 67.
 at ./bin/shutter line 10986.
```

Any ideas?

### CODE REVIEW

File: share/shutter/resources/modules/Shutter/Pixbuf/Save.pm [hunk 1]
```diff
@@ -49,17 +49,41 @@ sub new {
...
 	$self->{_lp}      = Shutter::Pixbuf::Load->new($self->{_common}, $current_window);
+	$self->{_quality} = undef;
 
 	bless $self, $class;
 	return $self;
 }
 
+sub set_quality_setting {
+		my $self = shift;
+		my $filetype = shift;
```
Comment (DarthGandalf) [inline share/shutter/resources/modules/Shutter/Pixbuf/Save.pm:60]: This accepts two parameters

File: share/shutter/resources/modules/Shutter/Pixbuf/Save.pm [hunk 2]
```diff
@@ -116,56 +131,29 @@ sub save_pixbuf_to_file {
...
-		if (my $settings = $self->{_common}->get_globalsettings_object) {
-			if (defined $settings->get_image_quality("png")) {
-				$quality = $settings->get_image_quality("png");
-			} else {
-				$quality = 9;
-			}
-		} else {
-			$quality = 9;
-		}
+		set_quality_setting($filetype);
```
Comment (DarthGandalf) [inline share/shutter/resources/modules/Shutter/Pixbuf/Save.pm:134]: You're passing a single parameter, so `$filetype` becomes `$self`.

Call this function as a method instead

File: share/shutter/resources/modules/Shutter/Pixbuf/Save.pm (hunk 1 above)
Comment (Photon89) [reply share/shutter/resources/modules/Shutter/Pixbuf/Save.pm:60]: Sorry, got it, I should call it with `$self->set_quality_setting`, not just `set_quality_setting`...

File: share/shutter/resources/modules/Shutter/Pixbuf/Save.pm (hunk 2 above)
Comment (Photon89) [reply share/shutter/resources/modules/Shutter/Pixbuf/Save.pm:134]: Right, thanks!

**Pull request closed without merging**